tools/modify_access.php

```diff
@@ -145,12 +142,11 @@
 {
     echo "<br>\n";
     echo "$action_type $activity_id ...<br>\n";
-    $yesno = ( $action_type == 'grant' ? 'yes' : 'no' );
 
-    $userSettings =& Settings::get_Settings($subject_username);
-    $userSettings->set_value("$activity_id.access", $yesno);
-
-    log_access_change( $subject_username, $pguser, $activity_id, $action_type );
+    if($action_type == 'grant')
+        $user->grant_access("$activity_id.access", $pguser);
+    else
+        $user->revoke_access("$activity_id.access", $pguser);
```

Comment:
{grant/revoke/request}_access should be invoked with $activity_id, not "$activity_id.access".

$activity_id is for the access_log table; "$activity_id.access" is for the usersettings table.

Reply:
<sigh> You're right, it is for these but not for the things in `manage_site_access_privileges.php` because They're Different. I'll have to noodle this over.

Reply:
Ok, I've fixed this in 4aefbc1.